### PR TITLE
feat: profiles search

### LINKED ISSUE #10448: Things needed to search profiles
image

question:
**what do we need to have the capability to search profiles**

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [Koda - Generative Art Marketplace](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Feature

## Needs Design check

- @exezbcz please review

## Needs QA check

- @kodadot/qa-guild please review

## Context

- [x] Closes #10448
- [x] Requires deployment https://github.com/kodadot/private-workers/pull/197

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

File: services/profile.ts [hunk 1]
```diff
@@ -104,6 +104,12 @@ export const fetchProfileByAddress = (address: string) =>
     method: 'GET',
   })
 
+export const searchProfiles = (query: string, limit = 5, offset = 0) =>
+  api<{ data: Profile[] }>('/profiles/search?q=query)}', {
```
Comment (preschian) [inline services/profile.ts:108]:
```suggestion
  api<{ data: Profile[] }>('/profiles/search)}', {
```

**Review by preschian — APPROVED** [review]

File: components/search/SearchProfiles.vue [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+const { isEvm } = useIsChain(urlPrefix)
+
+const emit = defineEmits(['close'])
+
+const { data: result, isLoading } = useQuery<{ data: Profile[] } | null>({
+  queryKey: ['search-profiles', computed(() => props.name)],
+  queryFn: () => (props.name ? searchProfiles(props.name, 10) : null),
+})
+
+const profileSuggestion = computed(() => result.value?.data || [])
```
Comment (hassnian) [inline components/search/SearchProfiles.vue:59]: let's make it `profileSuggestion` plural as is an array

would it be cleaner if we used this in `useQuery`? 

CleanShot 2024-07-09 at 09 47 40@2x

File: components/search/SearchProfiles.vue [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    type: String,
+    required: true,
+  },
+})
+
+const router = useRouter()
+const { urlPrefix } = usePrefix()
+const { isEvm } = useIsChain(urlPrefix)
+
+const emit = defineEmits(['close'])
```
Comment (hassnian) [inline components/search/SearchProfiles.vue:52]:
```suggestion
const emit = defineEmits(['close'])
const props = defineProps({
  name: {
    type: String,
    required: true,
  },
})

const router = useRouter()
const { urlPrefix } = usePrefix()
const { isEvm } = useIsChain(urlPrefix)
```

File: components/search/SearchProfiles.vue [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    <SearchResultItem v-for="item in 5" :key="item" is-loading />
+  </div>
+  <div v-else-if="!profileSuggestion.length" class="mx-6 mt-4">
+    {{ $t('search.profileNotFound', [name]) }}
+  </div>
+  <div v-else>
+    <div
+      v-for="item in profileSuggestion"
+      :key="item.address"
+      :value="item"
```
Comment (hassnian) [inline components/search/SearchProfiles.vue:12]: ?

```suggestion
```

**Comment by Jarsen136** [comment]
@hassnian All have been updated

File: components/search/SearchProfiles.vue (hunk 2 above)
Comment (Jarsen136) [reply components/search/SearchProfiles.vue]: There is a type issue here so I use another workaround instead

File: components/search/SearchProfiles.vue (hunk 4 above)
Comment (hassnian) [reply components/search/SearchProfiles.vue]: why is this value needed ?

Comment (Jarsen136) [reply components/search/SearchProfiles.vue]: removed already from the commit

**Review by hassnian — APPROVED** [review]

**Comment by Jarsen136** [comment]
image

I would deploy it https://github.com/kodadot/private-workers/pull/197

**Comment by vikiival** [comment]
@kodadot/internal-dev can you please deploy the profile worker ?

**Comment by Jarsen136** [comment]
> @kodadot/internal-dev can you please deploy the profile worker ?

I have deployed it

image

**Pull request merged**